oauthproxy.go

```diff
@@ -393,6 +395,14 @@ func (p *OAuthProxy) RobotsTxt(rw http.ResponseWriter) {
 
 // ErrorPage writes an error response
 func (p *OAuthProxy) ErrorPage(rw http.ResponseWriter, code int, title string, message string) {
+	if p.APIMode {
+		rw.Header().Set("Content-Type", applicationJSON)
```

Comment:
Should we just extend the functionality implemented in `ErrorJSON` with the improvements here and call that here to unify things?